tests/integration/suite/test_workloads.py

```diff
@@ -341,3 +369,19 @@ def wait_for_service_cluserip_reset(client, name, timeout=30):
         if time.time() - start > timeout:
             raise Exception('Timeout waiting for workload service')
     return services.data[0]
+
+
+def wait_for_workload_active(client, workload, timeout=30):
```

Comment:
This is a duplication of code that already lives in `wait_for`, use that and pass in your callback and fail handler.